docs/testing.md

```diff
@@ -1 +1,109 @@
+
+There are a large number of Matcher functions available for most cases, and they can be prepended with a ```.not.``` (e.g. ```expect(false).not.toBe(true)```) You can see these (with examples!) in the Jasmine Documentation [here](http://jasmine.github.io/2.3/introduction.html#section-Included_Matchers)
+
+Any variables or constants you might need across tests can be defined outside. The scoping rules mean that each spec has access to variables outside of them.
+
+```
+'use strict';
+define(['kb.config'], function(Config) {
+    var wsUrlConfig = 'services.workspace.url';
+    var correctWsUrl = 'https://kbase.us/services/ws';
+
+    describe('The KBase Config object', function() {
+        it('should have a Workspace url', function() {
+            expect(Config.getItem(wsUrlConfig)).toBe(correctWsUrl);
+        });
+    });
+});
+```
+
+That's pretty much it. You can certainly do more complex things, like simulate click events and DOM modification, but this is intended for an introduction. There are a few test suites available in test/specs, and more will come.
+
+# Karma for running tests
+
+Now that you have a number of tests written, they need to be run! The tool we use for running tests is called Karma [see homepage](http://karma-runner.github.io/0.13/index.html)
+
+Karma is a test running framework - that means it handles starting up a web server, web browser, and populates the test page with all necessary files, then runs the test. **This is already configured for you.** There's nothing you should need to do here, but this part of the document describes the process used.
+
+Karma uses a common conf.js file (commonly `karma.conf.js`). For us, that lives inside the test directory. This configuration file tells Karma what files need to be included in the web server, whether they need to be automatically loaded as part of the test page, and where the test files are. Fortunately, most of this is templateable. However, since most of our external components are installed by Bower, and that can lead to some inconsistencies in what gets installed, we had to include most of those manually. If you need to include a new external dependency, you'll likely have to do the same. Otherwise, all .js files in the src directory are included, and .js files under test/specs are treated as tests.
+
+The configuration is currently set to just test against PhantomJS - a headless, WebKit scriptable browser. Version of Chrome and Firefox will eventually be added to the testers, especially for automation on Travis-CI.
+
+Finally, Karma is configured to automatically report test coverage. When running locally using the commands below, an HTML directory with test results and coverage will be automatically generated for you to browse. When run on Travis-CI, a different file is made that gets passed along to Coveralls.
```

Comment:
The test coverage directory shows up in (from the root of ui-common) build/test-coverage/html/ It's referenced below, but kind of implicitly. I'll make it more obvious there.